changed: added geocoding api

roemhildtg opened the pull request:
In response to #123 

I created a service on /api/v2/geocoder.

It returns basic geojson with the lat/lon and Metadata from the provider (currently only mapbox)

New providers can be added without major rework. 

Note, additional config needed:

```js

    "geocoder": {
        "provider": "mapbox",
        "url": "https://api.mapbox.com/geocoding/v5/mapbox.places/{address}.json", // optional
        "apiKey": "pk....." // required
    },
```

Let me know your thoughts

MartijnR commented:
Thanks! Much appreciated. I haven't yet looked at the code, but just have a few initial comments:

1. I think this should not be under /api/v2 and won't be documented in https://apidocs.enketo.org/v2 since that API is for other apps to communicate with Enketo. In this case, it's an internal API. How about a geo-controller.js that responds to /geo/geocoder? 
2. In order to merge this we'd need to add the Google Geocoding service as well. Note that some installations using only Mapbox/Leaflet tile layers and Google Geocoding do/could still work at the moment. If you'd have the time to do so, that would be great. Otherwise, it will have to wait until I have time.
3. Please add a default value in default-config.json, defaulting to Google (to minimize required config changes for existing installations) when upgrading. 
4. In addition, to avoid problems with existing installations, I think we should use the google['api key'] as a backup if geocoder['api key'] is not available but google is selected as the geocoding service (which would be the default)
5. Please use the existing naming convention for config items (so "api key" instead of "apiKey")
6. Please document new config items in tutorials/10-configure.md

lognaturel commented:
Closing for now but please re-open if this becomes active again. Thanks!

roemhildtg commented:
Hi @lognaturel I have updated this merge request with @MartijnR 's feedback and I think its in a better place. Would you be open to reviewing it again? I've also updated the enketo core MR as well.
